### PR TITLE
Load animation in piskel

### PR DESCRIPTION
**Pull request by islemaster**
Upon selecting an animation in the Animation Tab, load that animation for editing in Piskel.

out

This updates our Piskel version because it requires the new message-receiving feature in [@code-dot-org/piskel v0.2.0](https://github.com/code-dot-org/piskel/releases/tag/v0.2.0).  It also wraps embedded Piskel in a `PiskelEditor` component that will own interactions with the frame.

### CODE REVIEW

File: apps/src/gamelab/GameLab.js [hunk 1]
```diff
@@ -238,13 +238,12 @@ GameLab.prototype.init = function (config) {
...
   var state = {};
-  var boundOnIsRunningChange = this.onIsRunningChange.bind(this);
-  store.subscribe(function () {
+  store.subscribe(() => {
     var lastState = state;
     state = store.getState();
 
     if (!lastState.runState || state.runState.isRunning !== lastState.runState.isRunning) {
-      boundOnIsRunningChange(state.runState.isRunning);
+      this.onIsRunningChange(state.runState.isRunning);
```
Comment (islemaster) [inline apps/src/gamelab/GameLab.js:246]: This bit of cleanup looks unrelated - I just happened to touch this method in an intermediate state where Gamelab.js was involved in communication with the iframe.  See commit-by-commit changes if you're curious.

File: apps/src/gamelab/AnimationTab/PiskelEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/** @file Component wrapping embedded Piskel editor */
+// PISKEL_DEVELOPMENT_MODE is a build flag.  See Gruntfile.js for how to enable it.
+/* global PISKEL_DEVELOPMENT_MODE */
+import React from 'react';
+import {connect} from 'react-redux';
+import { METADATA_SHAPE } from '../animationMetadata';
+import {MessageType} from '@code-dot-org/piskel';
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationTab/PiskelEditor.jsx:7]: Space on line 6 but not on lines 5 or 7? (My own preference is to have spaces, tho I think yours is usually not to)

File: apps/src/gamelab/AnimationTab/PiskelEditor.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    // Piskel should be hosted on the same origin (domain) as gamelab; we don't
+    // want to send messages to other domains.
+    // TODO (bbuchanan): Inject window origin to remove global dependency?
+    const targetOrigin = location.origin;
+    this.iframe.contentWindow.postMessage(message, targetOrigin);
+  },
+
+  render() {
+    return <iframe
+        ref={c => this.iframe = c}
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationTab/PiskelEditor.jsx]: Why `c`? Is this better than just doing ref="iframe" (and then `this.refs.iframe` in `postMessage`)

**Comment by Bjvanminnen** [comment]
couple nits, generally lgtm

File: apps/src/gamelab/AnimationTab/PiskelEditor.jsx (hunk 3 above)
Comment (islemaster) [inline apps/src/gamelab/AnimationTab/PiskelEditor.jsx]: Personally I think the string ref is simpler too, but I'm trying to stick to best practices.  From [The ref String Attribute](https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute):

> Although string refs are not deprecated, they are considered legacy, and will likely be deprecated at some point in the future. Callback refs are preferred.

And `c` because ~~I'm lazy~~ I stole it from the example on the page linked above.  Will fix.
